scripts/ci/run-zip-tests

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env python
+# Don't run tests from the root repo dir.
+# We want to ensure we're importing from the installed
+# binary package not from the CWD.
+
+import argparse
+import os
+import sys
+from contextlib import contextmanager
+from subprocess import check_call
+
+import botocore
+import zipfile
+import zipimport
+
+_dname = os.path.dirname
+
+REPO_ROOT = _dname(_dname(_dname(os.path.abspath(__file__))))
+PACKAGE = "botocore"
+
+
+@contextmanager
+def cd(path):
+    """Change directory while inside context manager."""
+    cwd = os.getcwd()
+    try:
+        os.chdir(path)
+        yield
+    finally:
+        os.chdir(cwd)
+
+
+def run(command):
+    return check_call(command, shell=True)
+
+
+def process_args(args):
+    runner = args.test_runner
+    test_args = ""
+    if args.import_mode_append:
+        test_args += "--import-mode=append "
+    dirs = " ".join(args.test_dirs)
+
+    return runner, test_args, dirs
+
+
+def check_state():
+    zip_loader = isinstance(botocore.__loader__, zipimport.zipimporter)
+    print(f'loader: {botocore.__loader__}')
+    zip_file = 'botocore.zip/botocore/__init__.py' in botocore.__file__
+    print(f'file: {botocore.__file__}')
+    zip_path = isinstance(botocore.BOTOCORE_ROOT, zipfile.Path)
+    print(f'type: {botocore.BOTOCORE_ROOT.__module__}')
+    zipped_state = (zip_loader, zip_file, zip_path)
+    if not all(zipped_state):
+        print("Botocore is not loaded in a zipped context")
+        print(os.environ['GITHUB_WORKSPACE'])
+        print(os.listdir())
+        sys.exit(1)
```

Comment:
It may be easier to read if the prints and variables are grouped together. The final failed state output may be able to be included in a single print statement for legibility.

Reply:
i actually included all of these print statements solely for debugging purposes.  zip test runs are failing on gh actions here and i'm trying to understand why. i've tested the workflow locally using `act` and it was working as expected